logs.go

```diff
@@ -91,6 +92,23 @@ func getLoggableResources(getPodContainers func(string, string) ([]string, error
 	return loggableResources, nil
 }
 
+// GetProjectPodsBySubstr returns a list of pod names that contain the provided string, in the named project.
+func getProjectPodNamesBySubstr(project, substr string) ([]string, error) {
+	ret := []string{}
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

And if we are to keep this function, we could organize it better to address the readability concerns:

```go
pods, err := GetResourceNames(project, "pod")
if err != nil {
	return nil, err
}
var filtered []string
for _, pod := range pods {
// ...
}
return filtered, nil
```

We only need `filtered` past some point into the function body. And perhaps `filtered` helps understand the intent better than `ret`?

And since we are talking about filtering, sharing a trick:
https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating

Anyway, I think this function would be easier to use if it didn't know about pods... you just need to filter an arbitrary slice of strings.